File: 1POO/Semestre 1 - 2/S2P1_setter_getter/P1A11Termometro/C++/Origen/Termometro.cpp
#include "Termometro.h"
#include <iostream>
#include <cstdlib>

using namespace std;

// Constructor por defecto
Termometro::Termometro() {
    temperatura = static_cast<float>(rand()) / RAND_MAX * 100;
}

// Constructor parametrizado
Termometro::Termometro(float temp) {
    temperatura = temp;
}

float Termometro::temperaturaC() {
    return temperatura;
}

float Termometro::temperaturaF() {
    return (temperatura * 9/5) + 32;
}

float Termometro::temperaturaK() {
    return temperatura + 273.15f;
}

void Termometro::mostrarTemperaturas() {
    cout << "Temperatura en Celsius: " << temperaturaC() << endl;
    cout << "Temperatura en Fahrenheit: " << temperaturaF() << endl;
    cout << "Temperatura en Kelvin: " << temperaturaK() << endl;
}

File: 1POO/Semestre 1 - 2/S2P1_setter_getter/P1A11Termometro/JAVA/Termometro.java
package P1A11Termometro.JAVA;

import java.util.Random;

public class Termometro {

    // Atributo
    private float temperatura;

    // Constructor por defecto
    public Termometro() {
        Random rand = new Random();
        this.temperatura = rand.nextFloat() * 100; // El random va del 0 al 100
    }

    // Constructor parametrizado (recibe temperatura en °C)
    public Termometro(float temperatura) {
        this.temperatura = temperatura;
    }

    // Temperatura en Celcius
    public float temperaturaC() {
        return temperatura;
    }

    // Temperatura en F
    public float temperaturaF() {
        return (temperatura * 9 / 5) + 32;
    }

    // Temperatura en K
    public float temperaturaK() {
        return (temperatura + 273.15f);
    }

    public void mostrartemperaturas() {
        System.out.println("Temperatura en Celcius: " + temperaturaC());
        System.out.println("Temperatura en Fahrenheit: " + temperaturaF());
        System.out.println("Temperatura en Kelvin: " + temperaturaK());
    }

}